satpy/readers/netcdf_utils.py

```diff
@@ -431,3 +434,205 @@ def _get_attr(self, obj, key):
         if self._use_h5netcdf:
             return obj.attrs[key]
         return super()._get_attr(obj, key)
+
+
+class Preloadable:
+    """Mixin class for pre-loading.
+
+    This is a mixin class designed to use with file handlers that support
+    preloading.  Subclasses deriving from this mixin are expected to be
+    geo-segmentable file handlers, and can be created based on a glob pattern
+    for a non-existing file (as well as a path or glob pattern to an existing
```

Comment:
Should this class be called eg `PreloadableSegments` to make clear of the main use case?

Reply:
Fixed in d362ff919